components/Ink/src/private/MDCInkLayer.m

```diff
@@ -15,590 +15,194 @@
  */
 
 #import "MDCInkLayer.h"
-#import "MDCInkLayer+Testing.h"
+#import "MaterialMath.h"
 
-#import <UIKit/UIKit.h>
+static const CGFloat MDCInkLayerCommonDuration = 0.083f;
+static const CGFloat MDCInkLayerStartScalePositionDuration = 0.333f;
+static const CGFloat MDCInkLayerStartFadeHalfDuration = 0.167f;
+static const CGFloat MDCInkLayerStartFadeHalfBeginTimeFadeOutDuration = 0.25f;
+static const CGFloat MDCInkLayerStartTotalDuration = 0.5f;
```

Comment:
Consider making use of the MotionInterchange to encode your motion timing information in the same format we're employing elsewhere in a follow-up PR. This will also enable the ink motion to be configured later on.

Reply:
Bonus: The MotionAnimator also makes it easy to interrupt explicit CALayer-based animations so that you can change the animation dimensions mid-flight.

Reply:
Yes, good idea. Will work on a follow-up PR once finishing this work.